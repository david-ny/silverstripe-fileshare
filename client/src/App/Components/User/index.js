import React from 'react';
import PropTypes from 'prop-types';
// import { UncontrolledDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
import {
  configPropType,
  configDefaultPropType,
} from '../../types';

const User = ({
  userID,
  firstName,
  lastName,
  email,
  config,
  setActAsUser,
}) => (
  <tr key={userID}>
    <td>
      <input type="checkbox" />
    </td>
    <td className="file">
      <button
        className="filename"
        alt={`${firstName} ${lastName}`}
        title={`${firstName} ${lastName}`}
        onClick={() => setActAsUser(userID, firstName, lastName, email)}
        type="button"
      >
        <img src={`${config.app.imagePath}/user.svg`} alt="folder" />
        {firstName} {lastName}
      </button>
    </td>
    <td />
    <td />
    <td>{email}</td>
    <td />
    <td>
      {/* <UncontrolledDropdown>
        <DropdownToggle caret className="file-menu-trigger">
          ...
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem>Edit</DropdownItem>
          <DropdownItem>Delete</DropdownItem>
        </DropdownMenu>
      </UncontrolledDropdown> */}
    </td>
  </tr>
);

User.propTypes = {
  config: configPropType,
  userID: PropTypes.number.isRequired,
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  email: PropTypes.string.isRequired,
  setActAsUser: PropTypes.func.isRequired,
};
User.defaultProps = {
  config: configDefaultPropType,
  firstName: '',
  lastName: '',
};

export default User;
