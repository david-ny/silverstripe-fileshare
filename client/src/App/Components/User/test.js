import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import User from './index';
import { configTestData } from '../../test/testData';

const props = configTestData;

describe('User', () => {
  it('renders without crashing', () => {
    const tbody = document.createElement('tbody');
    ReactDOM.render(<User {...props} />, tbody);
  });
  test('has a valid snapshot', () => {
    const component = renderer.create(
      <User {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
