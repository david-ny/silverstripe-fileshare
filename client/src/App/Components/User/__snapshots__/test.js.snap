// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User has a valid snapshot 1`] = `
<tr>
  <td>
    <input
      type="checkbox"
    />
  </td>
  <td
    className="file"
  >
    <button
      alt="John Doe"
      className="filename"
      onClick={[Function]}
      title="John Doe"
      type="button"
    >
      <img
        alt="folder"
        src="fileshare/client/dist/images/user.svg"
      />
      John
       
      Doe
    </button>
  </td>
  <td />
  <td />
  <td>
    john@test.com
  </td>
  <td />
  <td>
    <div
      className="dropdown"
      onKeyDown={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="file-menu-trigger dropdown-toggle btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        ...
      </button>
      <div
        aria-hidden={true}
        className="dropdown-menu"
        role="menu"
        tabIndex="-1"
      >
        <button
          className="dropdown-item"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          Edit
        </button>
        <button
          className="dropdown-item"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </td>
</tr>
`;
